lib/active_model/serializer.rb

```diff
@@ -164,6 +164,14 @@ def json_key
       end
     end
 
+    def id
+      object.id if object
+    end
+
+    def type
+      object.class.to_s.demodulize.underscore.pluralize
```

Comment:
1. I believe this should go into the adapter because it's not really used by anyone else and it's a JSONAPI need.
2. Having this in the serializer, we're forced to keep it public, which I think doesn't make much sense.